packages/connect/src/core/index.ts

```diff
@@ -621,7 +644,7 @@ const onCall = async (context: CoreContext, message: IFrameCallMessage) => {
     device.on(DEVICE.SAVE_STATE, (state: string) => {
         // Persist internal state only in case of core in popup
         // Currently also only for webextension until we asses security implications
-        if (useCoreInPopup && env === 'webextension') {
+        if (useCoreInPopup && env === 'webextension' && origin) {
```

Comment:
this is because in the onCall fn, origin is assigned as follows
```
    const origin = DataManager.getSettings('origin')!;
```
right?

Reply:
Yes, in this method we're not doing that so it's possibly an undefined type.
Maybe it would be better to check that in onCall as well.